motr/ut/obj.c

```diff
@@ -658,6 +658,35 @@ static void ut_test_entity_namei_op_delete(void)
 	ut_entity_namei_op(M0_EO_DELETE);
 }
 
+static void ut_test_obj_lid_assign(void)
+{
+	struct m0_obj     obj;
+	struct m0_entity  ent;
+	struct m0_realm   realm;
+	struct m0_op     *ops[1] = {NULL};
+	struct m0_client *instance = NULL;
+	int               rc;
+
+	/* init */
+	rc = ut_m0_client_init(&instance);
+	M0_UT_ASSERT(rc == 0);
+
+	ut_realm_entity_setup(&realm, &ent, instance);
+	ent.en_realm = &realm;
+
+	memset(&obj, 0, sizeof obj);
+	obj.ob_attr.oa_buf_size = 40960;
+	m0_obj_init(&obj, &realm, &ent.en_id, 0);
+
+	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
+	M0_UT_ASSERT(obj.ob_attr.oa_layout_id > 0 &&
+			obj.ob_attr.oa_layout_id < 15);
```

Comment:
Should we move this M0_UT_ASSERT() one line above to Line #680?
Because this is to check the result of m0_obj_init().

Reply:
I don't think that would work as the optimal lid is assigned during `m0_entity_create()` in `obj_op_obj_init()`
```
        if (OP_OBJ2CODE(oo) == M0_EO_CREATE) {
                rc = m0__obj_pool_version_get(obj, &pv);
                if (rc != 0)
                        return M0_ERR(rc);
                oo->oo_pver = pv->pv_id;

                ldom = &cinst->m0c_reqh.rh_ldom;
                obj_optimal_lid_set(obj, ldom);
```
So we need to check after `m0_entity_create()` is complete.

Reply:
OK